### ISSUE TITLE
Avoid duplicate publisher records by adding a global lookup check

### ISSUE THREAD

**Issue by SvenLieber** [kbrbe/beltrans-data-integration#62]
A publication MARC record of KBR contains a link to publishers in the field `264` as text and in the fields `710` as linked authority records (first investigated in this issue https://github.com/SvenLieber/beltrans-data/issues/34).

We cannot use the field `710` only, because some publishers are only mentioned as text in the field `264`, yet we also want to avoid duplicate mappings, where the same publisher with different names is mapped from `264` and `710`.
Therefore we implemented an algorithm based on string similarity and the levenshtein distance to identify which `264` text publisher corresponds to which `710` linked authority *within one MARC publication record*.
In case a publisher is only mentioned as text in the field `264` we create an own authority ID and map this publisher.

However, this results in newly created publisher IDs even though the publisher may already exist as linked authority within the KBR data (it's simply not used in the record thus we didn't find it).
Therefore we need an additional check for self created publisher records where we compare the name to *all authority records of KBR*.

**Comment by SvenLieber** [comment]
Just based on a simply lookup of normalized strings several self-made identifiers can be replaced with existing authorities found in the whole authority dump:

* 193 from 465 for the NL-FR translations could be replaced (38 were not replaced because several replacement candidates exist)
* 151 from 381 for the FR-NL translations could be replaced (53 were not replaced because several replacement candidates exist)

Several replacement candidates exist because different authorities have the same name. We get duplicate warnings for 5.7 percent of authority records.

**Comment by SvenLieber** [comment]
The devised solution still does not resolve all duplicates, for example we get the following duplicates were name disambiguity is only resolved sometimes:

* A. W. Bruna & Zoon (b61d9cef1d3d5bfac3f2dc815e93be9a)
* A. W. Bruna & zoon (b61d9cef1d3d5bfac3f2dc815e93be9a)
* A. W. Bruna (a98ef9f1d121a1abb64f4cf414095a8d)
* A.W. Bruna & zoon (2cd0029a9fffdd3152aed8546d719859)
* A.W. Bruna & Zoon (2cd0029a9fffdd3152aed8546d719859)
* A.W. Bruna& Zoon (aeac7a00ea956899037f89bd94ca0b51)
* Bruna & Zoon (245a340065ecdd9c335d94acfef4a316)
* Bruna & zoon (245a340065ecdd9c335d94acfef4a316)
* A.W. Bruna & Zoon (14376135)
* Bruna [A.W.] & Zoon (13961264)

In this case all the records should be linked to one of the last two *actual* authority records from the data source.

**Comment by SvenLieber** [comment]
In the shown case the double mentioning of publishers was fixed in https://github.com/kbrbe/beltrans-data-integration/issues/81

Since an automatic string comparison may always fail we added a step to the pipeline to support the global lookup. In a CSV file an existing (wrong) spelling of a publisher can be given in the first column and the spelling of the publisher name as found in the KBR authority records in the second column.

This additional step is performed before the lookup in the deduplication step,  no programming skills are needed as only names have to be provided in a CSV https://github.com/kbrbe/beltrans-data-integration/commit/f7916602f1597e77f56179ba26ba135d2bf118da

However, in the shown case of Bruna & Zoon, the underlying issue is that the spellings are found in the KBR authority records, but more than one are found which prevents us from take an (semi) automatic decision for all such records in the corpus.
